swap_init.py

```diff
@@ -181,27 +184,44 @@ def resolver_gen(nameservers):
         if ipv6_ns:
             valid_ns.append(random.choice(ipv6_ns))
 
-    resolv_data = '\n'.join("nameserver %s" %x for x in valid_ns)
-    resolv_data += '\noptions timeout:1 attempts:3 rotate\n'
+    if type == 'regular':
+        resolv_data = '\n'.join("nameserver %s" %x for x in valid_ns)
+        resolv_data += '\noptions timeout:1 attempts:3 rotate\n'
+    else:
+        resolv_data = ' '.join(valid_ns)
 
     return resolv_data
 
 
-@ifon('Linux', 'SunOS')
+@ifon('Linux')
 def resolver_setup(nameservers):
     """Writes resolv.conf file.
     """
 
-    with open("/etc/resolv.conf", 'w') as f:
-        f.write(resolver_gen(nameservers))
+    # regular resolv.conf
+    if not os.path.islink('/etc/resolv.conf'):
+        with open("/etc/resolv.conf", 'w') as f:
+            f.write(resolver_gen(nameservers, 'regular'))
+    # regular resolv.conf for resolvconf tool
     rconfdir='/etc/resolvconf/resolv.conf.d'
     if os.path.exists(rconfdir):
         with open('%s/original' % rconfdir, 'w') as f:
-                f.write(resolver_gen(nameservers))
+                f.write(resolver_gen(nameservers, 'regular'))
         rlink = '%s/tail' % rconfdir
         if os.path.exists(rlink):
             os.unlink(rlink)
         os.symlink('%s/original' % rconfdir, rlink)
+    # systemd-resolved
+    if os.path.exists('/etc/systemd/resolved.conf'):
+        dns_list=resolver_gen(nameservers, 'dnslist')
+        rpath='/etc/systemd/resolved.conf.d'
+        if not os.path.exists(rpath):
+            os.mkdir(rpath, 0o755)
+        with open('%s/gandi.conf' % rpath, 'w') as f:
+            f.write('[Resolve]\nDNS=%s\n' % dns_list)
+        # XXX we may have to move this to systemd-networkd in the futur
+        subprocess.Popen(['/bin/ip', '-4', 'addr', 'add', '127.0.0.53/32', 'dev', 'lo']).wait()
```

Comment:
A manual test creating `/etc/systemd/resolved.conf.d/gandi.conf` followed by a restart of `systemd-resolved` seems to indicate that this is not needed.

Reply:
Indeed, after another test on my setup, I removed the ip addr command and the two lines in /etc/network/interfaces.